fix TestWaitForInterrupt test

i see that this test sometimes fails on [CI](https://circleci.com/gh/mesg-foundation/core/3736?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link):
```
--- FAIL: TestWaitForInterrupt (1.00s)
	signal_test.go:32: timeout waiting for signal
FAIL
coverage: 100.0% of statements
FAIL	github.com/mesg-foundation/core/x/xsignal	1.005s
```

--- a/x/xsignal/signal_test.go
+++ b/x/xsignal/signal_test.go
@@ -14,26 +14,22 @@ func TestWaitForInterrupt(t *testing.T) {
 	defer signal.Stop(abort)
 
 	syscall.Kill(os.Getpid(), syscall.SIGINT)
-	syscall.Kill(os.Getpid(), syscall.SIGTERM)
-
-	sigCount := 0
-
-loop:
-	for {
-		select {
-		case sig := <-abort:
-			if sig != syscall.SIGTERM && sig != syscall.SIGINT {
-				t.Fatalf("unexpected signal - got %v", sig)
-			}
-			if sigCount = sigCount + 1; sigCount >= 2 {
-				break loop
-			}
-		case <-time.After(1 * time.Second):
-			t.Fatalf("timeout waiting for signal")
+	select {
+	case sig := <-abort:
+		if sig != syscall.SIGINT {
+			t.Fatalf("unexpected signal - got %v, want %v", sig, syscall.SIGINT)
 		}
+	case <-time.After(1 * time.Second):
+		t.Fatalf("timeout waiting for signal")
 	}
 
-	if sigCount != 2 {
-		t.Fatalf("received signals failed: got: %d, want: %d", sigCount, 2)
+	syscall.Kill(os.Getpid(), syscall.SIGTERM)
+	select {
+	case sig := <-abort:
+		if sig != syscall.SIGTERM {
+			t.Fatalf("unexpected signal - got %v, want %v", sig, syscall.SIGTERM)
+		}
+	case <-time.After(1 * time.Second):
+		t.Fatalf("timeout waiting for signal")
 	}
 }

